src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Public.Data/DataSetsControllerTests.cs

```diff
@@ -952,4 +963,111 @@ private async Task<HttpResponseMessage> GetDataSet(Guid dataSetId, HttpClient? c
             return await client.GetAsync(uri);
         }
     }
+
+    public class CreateDataSetVersionTests(TestApplicationFactory testApp) : DataSetsControllerTests(testApp)
+    {
+        public static IEnumerable<object[]> AllDataSetVersionStatuses =>
+            EnumUtil.GetEnums<DataSetVersionStatus>()
+            .Select(e => new object[] { e });
+
+        [Fact]
+        public async Task Success()
+        {
+            var dataSetVersionId = Guid.NewGuid();
+            var releaseFileId = Guid.NewGuid();
+
+            var processorClient = new Mock<IProcessorClient>();
+            processorClient
+                .Setup(c => c.Process(
+                    releaseFileId,
+                    It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new ProcessorTriggerResponseViewModel
+                {
+                    DataSetVersionId = dataSetVersionId,
+                    InstanceId = Guid.NewGuid()
+                });
+
+            var client = BuildApp(processorClient.Object).CreateClient();
+
+            var response = await CreateDataSetVersion(releaseFileId, client);
+
+            var content = response.AssertOk<DataSetVersionCreateViewModel>();
+
+            Assert.NotNull(content);
+            Assert.Equal(dataSetVersionId, content.DataSetVersionId);
+        }
+
+        [Fact]
+        public async Task NotBauUser_Returns403()
+        {
+            var client = BuildApp(user: AuthenticatedUser()).CreateClient();
+
+            var response = await CreateDataSetVersion(Guid.NewGuid(), client);
+
+            response.AssertForbidden();
+        }
+
+        [Theory]
+        [MemberData(nameof(AllDataSetVersionStatuses))]
+        public async Task DataSetVersionExistsForReleaseFile_Returns400(DataSetVersionStatus dataSetVersionStatus)
+        {
+            var releaseFileId = Guid.NewGuid();
+
+            DataSet dataSet = DataFixture
+                .DefaultDataSet()
+                .WithStatusPublished();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            DataSetVersion dataSetVersion = DataFixture
+                .DefaultDataSetVersion(
+                    filters: 1,
+                    indicators: 1,
+                    locations: 1,
+                    timePeriods: 2)
```

Comment:
There's been a recent change to make the data set version meta optional now. I think we could take advantage of that here and remove specifying these options as the meta is insignificant to this test.